### PR TITLE
[RBAC] Support grant role and revoke role

### PR DESCRIPTION
**Pull request by Junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR support grant role and revoke role statement for RBAC.

## Changelog

- New Feature
## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/4201 https://github.com/datafuselabs/databend/issues/2818

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by flaneur2020** [comment]
/lgtm

nice work!  :heart:

**Comment by BohuTANG** [comment]
Oops, conflicts

File: query/src/interpreters/interpreter_role_grant.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Datafuse Labs.
```
Comment (youngsofun) [inline query/src/interpreters/interpreter_role_grant.rs:1]: 2022

Comment (Junnplus) [reply query/src/interpreters/interpreter_role_grant.rs:1]: We can consider removing this time uniformly from license, should be in another PR.

File: query/src/interpreters/interpreter_role_drop.rs [hunk 2]
```diff
@@ -52,7 +52,7 @@ impl Interpreter for DropRoleInterpreter {
         let tenant = self.ctx.get_tenant();
         let user_mgr = self.ctx.get_user_manager();
         user_mgr
-            .drop_role(&tenant, &plan.role_identity, plan.if_exists)
+            .drop_role(&tenant, plan.role_identity, plan.if_exists)
```
Comment (youngsofun) [inline query/src/interpreters/interpreter_role_drop.rs:55]: do we need to check the role is used(granted) to other user/role?

**Review by youngsofun — APPROVED** [review]

**Pull request merged**